lib/class-wp-rest-widgets-controller.php

```diff
@@ -0,0 +1,695 @@
+	 *
+	 * @since 5.6.0
+	 *
+	 * @return array[]
+	 */
+	public function get_collection_params() {
+		return array(
+			'context' => $this->get_context_param( array( 'default' => 'view' ) ),
+			'sidebar' => array(
+				'description' => __( 'The sidebar to return widgets for.', 'gutenberg' ),
+				'type'        => 'string',
+			),
+		);
+	}
+
+	/**
+	 * Retrieves the widget's schema, conforming to JSON Schema.
+	 *
+	 * @since 5.6.0
+	 *
+	 * @return array Item schema data.
+	 */
+	public function get_item_schema() {
+		if ( $this->schema ) {
+			return $this->schema;
+		}
+
+		$this->schema = array(
+			'$schema'    => 'http://json-schema.org/draft-04/schema#',
+			'title'      => 'widget',
+			'type'       => 'object',
+			'properties' => array(
+				'id'            => array(
+					'description' => __( 'Unique identifier for the widget.', 'gutenberg' ),
+					'type'        => 'string',
+					'context'     => array( 'view', 'edit', 'embed' ),
+				),
+				'id_base'       => array(
+					'description' => __( 'Type of widget for the object.', 'gutenberg' ),
+					'type'        => 'string',
+					'context'     => array( 'view', 'edit', 'embed' ),
+				),
+				'sidebar'       => array(
+					'description' => __( 'The sidebar the widget belongs to.', 'gutenberg' ),
+					'type'        => 'string',
+					'default'     => 'wp_inactive_widgets',
+					'required'    => true,
+					'context'     => array( 'view', 'edit', 'embed' ),
+				),
+				'widget_class'  => array(
+					'description' => __( 'Class name of the widget implementation.', 'gutenberg' ),
+					'type'        => 'string',
+					'context'     => array( 'view', 'edit', 'embed' ),
+				),
+				'name'          => array(
+					'description' => __( 'Name of the widget.', 'gutenberg' ),
+					'type'        => 'string',
+					'context'     => array( 'view', 'edit', 'embed' ),
+				),
+				'description'   => array(
+					'description' => __( 'Description of the widget.', 'gutenberg' ),
+					'type'        => 'string',
+					'context'     => array( 'view', 'edit', 'embed' ),
+				),
+				'number'        => array(
+					'description' => __( 'Number of the widget.', 'gutenberg' ),
+					'type'        => 'integer',
+					'context'     => array( 'view', 'edit', 'embed' ),
+				),
+				'rendered'      => array(
+					'description' => __( 'HTML representation of the widget.', 'gutenberg' ),
+					'type'        => 'string',
+					'context'     => array( 'view', 'edit', 'embed' ),
+					'readonly'    => true,
+				),
+				'rendered_form' => array(
+					'description' => __( 'HTML representation of the widget admin form.', 'gutenberg' ),
+					'type'        => 'string',
+					'context'     => array( 'edit' ),
+					'readonly'    => true,
+				),
+				'settings'      => array(
+					'description' => __( 'Settings of the widget.', 'gutenberg' ),
+					'type'        => 'object',
+					'context'     => array( 'view', 'edit', 'embed' ),
+					'default'     => array(),
+				),
+			),
```

Comment:
I think it would be useful to add `widget_options`
https://github.com/WordPress/wordpress-develop/blob/cfc3b57488458801c64c750cb500c98c1ef635ad/src/wp-includes/class-wp-widget.php#L166-L172

Specially `customize_selective_refresh`.

Reply:
Agreed. We can add this when the widget type endpoint is implemented. No need to add it here now and just remove it in a few days I think.